### PR TITLE
upgrade: Rust 1.49.0

### PR DESCRIPTION
**Pull request by bartlomieju**

### CODE REVIEW

File: core/async_cancel.rs [hunk 1]
```diff
@@ -334,7 +334,7 @@ mod internal {
 
   impl PartialEq for Node {
     fn eq(&self, other: &Self) -> bool {
-      self as *const _ == other as *const _
+      std::ptr::eq(self, other)
```
Comment (bartlomieju) [inline core/async_cancel.rs:337]: @piscisaureus please take a look, clippy was complaining about previous version and suggested this instead

Comment (piscisaureus) [reply core/async_cancel.rs:337]: Looks good to me.

File: runtime/ops/fs.rs [hunk 2]
```diff
@@ -1660,15 +1660,14 @@ async fn op_utime_async(
   args: Value,
   _zero_copy: BufVec,
 ) -> Result<Value, AnyError> {
-  let state = state.borrow();
```
Comment (piscisaureus) [inline runtime/ops/fs.rs:1663]: Is this also necessitated by a clippy/lint change in rust 1.49? 
I'm curious what it is.

Comment (bartlomieju) [reply runtime/ops/fs.rs:1663]: Yes, clippy started complaining about about ref cells being borrowed across await points as well as mutex locks, eg:
```
error: this RefCell Ref is held across an 'await' point. Consider ensuring the Ref is dropped before calling await.
    --> runtime/ops/fs.rs:1663:7
     |
1663 |   let state = state.borrow();
     |       ^^^^^
     |
     = note: `-D clippy::await-holding-refcell-ref` implied by `-D clippy::all`
note: these are all the await points this ref is held through
    --> runtime/ops/fs.rs:1663:3
     |
1663 | /   let state = state.borrow();
1664 | |   super::check_unstable(&state, "Deno.utime");
1665 | |
1666 | |   let args: UtimeArgs = serde_json::from_value(args)?;
...    |
1679 | |   .unwrap()
1680 | | }
     | |_^
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_refcell_ref

error: this RefCell Ref is held across an 'await' point. Consider ensuring the Ref is dropped before calling await.
    --> runtime/ops/fs.rs:1663:7
     |
1663 |   let state = state.borrow();
     |       ^^^^^
     |
     = note: `-D clippy::await-holding-refcell-ref` implied by `-D clippy::all`
note: these are all the await points this ref is held through
    --> runtime/ops/fs.rs:1663:3
     |
1663 | /   let state = state.borrow();
1664 | |   super::check_unstable(&state, "Deno.utime");
1665 | |
1666 | |   let args: UtimeArgs = serde_json::from_value(args)?;
...    |
1679 | |   .unwrap()
1680 | | }
     | |_^
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_refcell_ref
```

```
    --> cli/module_graph.rs:2303:9
     |
2303 |     let h = handler.lock().unwrap();
     |         ^
     |
     = note: `-D clippy::await-holding-lock` implied by `-D clippy::all`
note: these are all the await points this lock is held through
    --> cli/module_graph.rs:2303:5
     |
2303 | /     let h = handler.lock().unwrap();
2304 | |     assert_eq!(h.version_calls.len(), 2);
2305 | |     let ver0 = h.version_calls[0].1.clone();
2306 | |     let ver1 = h.version_calls[1].1.clone();
...    |
2326 | |     assert!(h.version_calls[1].1 == ver0 || h.version_calls[1].1 == ver1);
2327 | |   }
     | |___^
     = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#await_holding_lock
```

Comment (piscisaureus) [reply runtime/ops/fs.rs:1663]: If clippy is complaining about borrowing a RefCell across a `.await`, I'd suggest to insert `drop(state)` before `tokio::task::spawn_blocking()`. It's a bit pedantic of me, but borrowing the same `RefCell` twice in a row for no clear reason doesn't pass the sniff test either.

**Review by piscisaureus — APPROVED** [review]
LGTM, but take a look at https://github.com/denoland/deno/pull/8955/files#r550813669.

**Comment by bartlomieju** [comment]
> LGTM, but take a look at https://github.com/denoland/deno/pull/8955/files#r550813669.

@piscisaureus I did that, but it still doesn't make clippy happy for some reason (which is a bit surprising).

**Pull request merged**